src/ML/src/MANNAutoregressive.cpp

```diff
@@ -0,0 +1,706 @@
+                                     const Contacts::EstimatedContact& rightFoot,
+                                     const manif::SE3d& basePosition,
+                                     const manif::SE3Tangentd& baseVelocity,
+                                     const AutoregressiveState& autoregressiveState,
+                                     const std::chrono::nanoseconds& time)
+{
+    constexpr auto logPrefix = "[MANNAutoregressive::reset]";
+
+    auto updateFoot = [logPrefix](const char* footName,
+                                  const Contacts::EstimatedContact& newContact,
+                                  Contacts::EstimatedContact& out) -> bool {
+        if (out.index != newContact.index)
+        {
+            log()->error("{} The index of the left {} is different from the one provided. "
+                         "Provided: {}, Expected: {}.",
+                         logPrefix,
+                         footName,
+                         newContact.index,
+                         out.index);
+            return false;
+        }
+        out = newContact;
+        return true;
+    };
+
+    if (!updateFoot("left", leftFoot, m_pimpl->output.leftFoot)
+        || !updateFoot("right", rightFoot, m_pimpl->output.rightFoot))
+    {
+        return false;
+    }
+
+    m_pimpl->mannInput = input;
+    m_pimpl->currentTime = time;
+    m_pimpl->isRobotStopped = true;
+
+    if (!m_pimpl->kinDyn.setRobotState(basePosition.transform(),
+                                       input.jointPositions,
+                                       baseVelocity.coeffs(),
+                                       input.jointVelocities,
+                                       m_pimpl->gravity))
+    {
+        log()->error("{} Unable to reset the kindyncomputations object.", logPrefix);
+        return false;
+    }
+
+    // the output is not valid since we are resetting the trajectory planner
+    m_pimpl->isOutputValid = false;
+    m_pimpl->supportCornerIndex = -1;
+    m_pimpl->supportFootPtr = nullptr;
+    m_pimpl->state = autoregressiveState;
+
+    if (!m_pimpl->baseOrientationDynamics->setState({basePosition.quat()}))
+    {
+        log()->error("{} Unable to reset the base orientation dynamics.", logPrefix);
+        return false;
+    }
+
+    if (!m_pimpl->mann.setInput(m_pimpl->mannInput))
+    {
+        log()->error("{} Unable to set the inputs to MANN.", logPrefix);
+        return false;
+    }
+
+    m_pimpl->fsmState = Impl::FSM::Reset;
+
+    return true;
+}
+
+bool MANNAutoregressive::setInput(const Input& input)
+{
+    // If the output of the network is valid, then the network has been called at least once
+    constexpr auto logPrefix = "[MANNAutoregressive::setInput]";
+
+    if (m_pimpl->fsmState != Impl::FSM::Reset && !this->isOutputValid())
+    {
+        log()->error("{} The network has not been reset. Please call reset().", logPrefix);
```

Comment:
Can't we do this automatically?

Reply:
Reset needs some information required only if we want to reset the network, for instance, the past trajectory. They should be provided by the user

Reply:
I see, nevermind then